src/components/input-time/input-time.tsx

```diff
@@ -0,0 +1,872 @@
+
+  @Watch("value")
+  valueWatcher(newValue: string): void {
+    if (!this.userChangedValue) {
+      this.setValue({ newValue, context: "direct" });
+    }
+    this.userChangedValue = false;
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  private activeEl: HTMLSpanElement;
+
+  defaultValue: InputTime["value"];
+
+  formEl: HTMLFormElement;
+
+  private hourEl: HTMLSpanElement;
+
+  labelEl: HTMLCalciteLabelElement;
+
+  private meridiemEl: HTMLSpanElement;
+
+  private meridiemOrder: number;
+
+  private minuteEl: HTMLSpanElement;
+
+  private previousEmittedValue: string;
+
+  previousValue: string;
+
+  private secondEl: HTMLSpanElement;
+
+  private userChangedValue = false;
+
+  // --------------------------------------------------------------------------
+  //
+  //  State
+  //
+  // --------------------------------------------------------------------------
+
+  @State() hour: string;
+
+  @State() hourCycle: HourCycle;
+
+  @State() localizedHour: string;
+
+  @State() localizedHourSuffix: string;
+
+  @State() localizedMeridiem: string;
+
+  @State() localizedMinute: string;
+
+  @State() localizedMinuteSuffix: string;
+
+  @State() localizedSecond: string;
+
+  @State() localizedSecondSuffix: string;
+
+  @State() meridiem: Meridiem;
+
+  @State() minute: string;
+
+  @State() second: string;
+
+  @State() showSecond: boolean = this.step < 60;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * Fires when the time value is committed.
+   */
+  @Event() calciteInputTimeChange: EventEmitter<string>;
+
+  /**
+   * Fires each time the user changes the value but has not committed changes.
+   */
+  @Event() calciteInputTimeInput: EventEmitter<string>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInternalInputTimeBlur: EventEmitter<void>;
+
+  /**
+   * @internal
+   */
+  @Event() calciteInternalInputTimeFocus: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Event Listeners
+  //
+  //--------------------------------------------------------------------------
+
+  @Listen("blur")
+  blurHandler(): void {
+    this.activeEl = undefined;
+    this.calciteInternalInputTimeBlur.emit();
+    this.emitChangeIfUserModified();
+  }
+
+  @Listen("click")
+  clickHandler(event: MouseEvent): void {
+    const composedEventPath = event.composedPath();
+    if (composedEventPath.includes(this.hourEl)) {
+      return;
+    }
+    if (composedEventPath.includes(this.minuteEl)) {
+      return;
+    }
+    if (composedEventPath.includes(this.secondEl)) {
+      return;
+    }
+    if (composedEventPath.includes(this.meridiemEl)) {
+      return;
+    }
```

Comment:
Can you use one if statement for all of these, it seems overly verbose. Maybe just a bunch of `||`'s, or throw the props in any array and use `some()`.

```js
if ([this.hourEl, this.minuteEl, this.secondEl, this.meridiemEl].some((el) =>
    composedEventPath.includes(el))) return;
```

`||` is better performance wise though

Reply:
the performance difference is very slight however so in this case I would defer to which seems more readable.